romanisim/bandpass.py

```diff
@@ -49,11 +49,17 @@ def read_gsfc_effarea(filename=None):
     astropy.table.Table
         table with effective areas for different Roman bandpasses.
     """
+
+    reader = functools.partial(Table.read, format='csv', delimiter=',',
+                               header_start=1, data_start=2)
     if filename is None:
-        dirname = pkg_resources.resource_filename('romanisim', 'data')
-        filename = os.path.join(dirname, 'Roman_effarea_20201130.csv')
-    return Table.read(filename, format='csv', delimiter=',', header_start=1,
-                      data_start=2)
+        ref = (resources.files('romanisim') / 'data'
+               / 'Roman_effarea_20201130.csv')
+        with resources.as_file(ref) as filename:
+            out = reader(filename)
```

Comment:
I am concerned about this breaking unexpectedly, due to the file being cleaned up with the `with` statement, as the reader is an iterator and not, say, a copy of the contents. I think we should use the `contextlib` stuff and control the lifetime of the temporary file.

Reply:
In this case reader is just Table.read(...)?  I would worry more if this were asdf with lazy loading, etc., but as far as I know that's not a concept with Table.read?

Reply:
Hrm.. python won't close it until the `out` object is deleted, it just seems poor form to have a "hidden" attachment there, but this will still run fine (assuming nothing tries to clear the temporary space containing the file). Maybe I am simply being too conservative.

Reply:
The desired behavior here is that out is an in-memory object that has all of the file contents and resources.as_file(ref) never needs to be seen again; I expect the temporary file to be cleaned up when it exits the with statement.  I just don't expect to need this temporary file again after we've read its contents into memory in the out object.  But okay, I'm passing tests here and I'm following the idiom at https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-filename closely, so I think I'm going to declare victory here.